feature/experimental/qos/ate_tests/mixed_sp_wrr_traffic_test/mixed_sp_wrr_traffic_test.go

```diff
@@ -0,0 +1,250 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mixed_sp_wrr_traffic_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/openconfig/featureprofiles/internal/deviations"
+	"github.com/openconfig/featureprofiles/internal/fptest"
+	"github.com/openconfig/ondatra"
+	"github.com/openconfig/ondatra/gnmi"
+	"github.com/openconfig/ondatra/gnmi/oc"
+	"github.com/openconfig/ygot/ygot"
+)
+
+type trafficData struct {
+	trafficRate float64
+	frameSize   uint32
+	dscp        uint8
+	queue       string
+	inputIntf   *ondatra.Interface
+}
+
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+// Test cases:
+//  - https://github.com/openconfig/featureprofiles/blob/main/feature/qos/ate_tests/mixed_sp_wrr_traffic_test/README.md
+//
+// Topology:
+//       ATE port 1
+//        |
+//       DUT--------ATE port 3
+//        |
+//       ATE port 2
+//
+//  Sample CLI command to get telemetry using gmic:
+//   - gnmic -a ipaddr:10162 -u username -p password --skip-verify get \
+//      --path /components/component --format flat
+//   - gnmic tool info:
+//     - https://github.com/karimra/gnmic/blob/main/README.md
+//
+
+func TestQoSCounters(t *testing.T) {
+	dut := ondatra.DUT(t, "dut")
+	dp1 := dut.Port(t, "port1")
+	dp2 := dut.Port(t, "port2")
+	dp3 := dut.Port(t, "port3")
+
+	// Configure DUT interfaces.
+	ConfigureDUTIntf(t, dut)
+
+	// Configure ATE interfaces.
+	ate := ondatra.ATE(t, "ate")
+	ap1 := ate.Port(t, "port1")
+	ap2 := ate.Port(t, "port2")
+	ap3 := ate.Port(t, "port3")
+	top := ate.Topology().New()
+	intf1 := top.AddInterface("intf1").WithPort(ap1)
+	intf1.IPv4().
+		WithAddress("198.51.100.1/31").
+		WithDefaultGateway("198.51.100.0")
+	intf2 := top.AddInterface("intf2").WithPort(ap2)
+	intf2.IPv4().
+		WithAddress("198.51.100.3/31").
+		WithDefaultGateway("198.51.100.2")
+	intf3 := top.AddInterface("intf3").WithPort(ap3)
+	intf3.IPv4().
+		WithAddress("198.51.100.5/31").
+		WithDefaultGateway("198.51.100.4")
+	top.Push(t).StartProtocols(t)
+
+	var trafficFlows map[string]*trafficData
+
+	switch dut.Vendor() {
```

Comment:
@sezhang2 why do we have more flows for ARISTA vs. Juniper and Cisco which appear to be the same except for queue assignment?   Different queue numberings per vendor seems ok?  But shouldn't the traffic on the wire be the same across all vendor cases?

Reply:
Yes, the traffic would same for all vendors based on our test plan.
  - https://github.com/openconfig/featureprofiles/blob/main/feature/qos/ate_tests/mixed_sp_wrr_traffic_test/README.md

Reply:
Ok, but the traffic flows are slightly different for arista for some reason?  trafficrate and framesize for example are not the same here:

https://github.com/openconfig/featureprofiles/pull/977/files/d6663d697873a778d75f3e2bddef48c8d9b5bc6e#diff-7ad47340a83785ccaedf86016c00d364191da968a47ee2a468cabda638549e23R92

as here:
https://github.com/openconfig/featureprofiles/pull/977/files/d6663d697873a778d75f3e2bddef48c8d9b5bc6e#diff-7ad47340a83785ccaedf86016c00d364191da968a47ee2a468cabda638549e23R107

Reply:
> Ok, but the traffic flows are slightly different for arista for some reason? trafficrate and framesize for example are not the same here:
> 
> https://github.com/openconfig/featureprofiles/pull/977/files/d6663d697873a778d75f3e2bddef48c8d9b5bc6e#diff-7ad47340a83785ccaedf86016c00d364191da968a47ee2a468cabda638549e23R92
> 
> as here: https://github.com/openconfig/featureprofiles/pull/977/files/d6663d697873a778d75f3e2bddef48c8d9b5bc6e#diff-7ad47340a83785ccaedf86016c00d364191da968a47ee2a468cabda638549e23R107

Ack. The traffic flow will have same values as described in the test plan in the next commit.